Code/Legacy/CrySystem/DebugCallStack.cpp

```diff
@@ -153,13 +154,13 @@ void DebugCallStack::SetUserDialogEnable(const bool bUserDialogEnable)
 DWORD g_idDebugThreads[10];
 const char* g_nameDebugThreads[10];
 int g_nDebugThreads = 0;
-volatile int g_lockThreadDumpList = 0;
+AZStd::spin_mutex g_lockThreadDumpList = 0;
 
 void MarkThisThreadForDebugging(const char* name)
 {
     EBUS_EVENT(AZ::Debug::EventTraceDrillerSetupBus, SetThreadName, AZStd::this_thread::get_id(), name);
 
-    WriteLock lock(g_lockThreadDumpList);
+    AZStd::lock_guard<AZStd::spin_mutex> lock(g_lockThreadDumpList);
```

Comment:
This can be changed to be
```suggestion
    AZStd::scoped_lock lock(g_lockThreadDumpList);
```